### DISCUSSION TITLE
Event Handler - how Conductor auto polling works?

### DISCUSSION THREAD

**Discussion by krishnapyde** [Netflix/conductor#2802] [Q&A] [answered]
I have tried the event handler as in the example here https://netflix.github.io/conductor/labs/eventhandlers/.
The WAIT task took almost 3 seconds to complete.

workflow 1 posted an event and went into WAIT mode. 
workflow 2 consumed that event and produced another event which the WAIT task in workflow 1 used to move to Completed status.

My question is, why did it take 3 seconds to complete this flow? 
My initial thoughts are that there is some kind of polling time set somewhere internally in conductor. (And i know that the default polling time if using a worked in 1second, but we are not using any worker here).
Can you help me with this?

Second, 
Is it correct to use this approach for a near real-time application?
i.e if I am to use this approach to orchestrate a near real-time flow,  (let's say after a financial transaction is initiated and before it is completed, I need to perform certain checks, and this needs to happen in near real-time, i.e. in a few milliseconds), How can I ensure that the time spent in orchestration is very minimal and does not add substantial time to the overall flow?

or should we stick to HTTP tasks for near real-time applications?

**Comment by v1r3n** [comment, marked as answer]
You want to stick the HTTP task for the near real time applications.
Conductor uses background threads and pollers for the event tasks which goes and marks WAIT tasks as completed and are useful in cases where the workflow keeps waiting for an external trigger and hence the integrations with various external eventing systems.

Would love to hear more about your use cases for near real-time applications and happy to provide guidance and suggestions on how to accomplish that. Certainly possible with Conductor.

**Reply by krishnapyde** [reply]
Thanks, @v1r3n . Your help/guidance will definitely help.
How do you propose we discuss this? Do you want me to prepare and send a doc with all the details and questions and you revert on that or can we do it over a call?

**Reply by v1r3n** [reply]
Happy to do this over the call if you prefer.  Alternatively we can also continue discussing here.
